src/vault/client/http.clj

```diff
@@ -239,14 +239,14 @@
 
 
   (read-secret
-    [this path opts]
+    [this path opts merge-req]
```

Comment:
> `merge-req`

this name is a little surprising to me, since it implies this is a function that will merge the request somehow.

Also, instead of a separate + new arity what about putting this map into `opts` under a suitable key?

Suggestions: `:request-opts`?

So line 249 would then look like:

```clojure
    (api-util/api-request this :get path (:request-opts opts))
```

Having been passed the `opts`

```clojure
{:request-opts
 {:query-params {"version" "version-xyz"}}
  ,,,}
```

Reply:
Good point! Passing it through opts (similar to the way I did in `kvv2.clj`) seems cleaner